src/EndcapTOF_geo.cpp

```diff
@@ -27,460 +27,224 @@ using namespace dd4hep::detail;
-  log_cooling_tube.setVisAttributes(description.visAttributes("TOFAluminum"));
-
-  std::string cooling_liquid_name = det_name + "_cooling_liquid_" + std::to_string(layer_id);
-  Tube        sol_cooling_liquid(0, cooling_tube_diameter / 2.0 - cooling_tube_thickness, baseplate_xdim / 2.0);
-  Volume      log_cooling_liquid(cooling_liquid_name, sol_cooling_liquid, description.material("Water"));
-  log_cooling_liquid.setVisAttributes(description.visAttributes("TOFWater"));
-
-  int isensor = 0;
-  // number of sensors+halfSH in y-radial direction
-  int nSensY = (int)((disk_rMax - (sensor_hybrid_ydim / 2)) / sensor_hybrid_ydim);
-
-  for (int ilay_y = nSensY; ilay_y >= -nSensY; ilay_y--) {
-    int nSensX =
-        (int)((2 * sqrt(pow(disk_rMax, 2) - pow((abs(ilay_y) * sensor_hybrid_ydim + sensor_hybrid_ydim / 2), 2))) /
-              baseplate_xdim);
-
-    if (nSensX == 0)
-      continue;
-    // we want an odd number of towers to be symmetrically centered around 0
-    if (nSensX % 2 == 0)
-      nSensX -= 1;
-
-    for (int ilay_x = -(nSensX - 1) / 2; ilay_x <= (nSensX - 1) / 2; ++ilay_x) {
-      if (ilay_x < 0) {
-        if (ilay_y < 0) {
-          if (sqrt(pow(ilay_x * baseplate_xdim + baseplate_xdim / 2 - disk_xOffset, 2) +
-                   pow(ilay_y * sensor_hybrid_ydim + sensor_hybrid_ydim / 2.0, 2)) < disk_rMin)
-            continue;
-        } else {
-          if (sqrt(pow(ilay_x * baseplate_xdim + baseplate_xdim / 2 - disk_xOffset, 2) +
-                   pow(ilay_y * sensor_hybrid_ydim - sensor_hybrid_ydim / 2.0, 2)) < disk_rMin)
-            continue;
+  int module = 0;
+  int nx     = 25;
+  int ny     = 15;
+  for (int ix = 0; ix < 2 * nx; ix++) {
+    float xcoord = (ix - nx) * (module_x + module_spacing);
+    for (int iy = 0; iy < 2 * ny; iy++) {
+      float ycoord = (iy - ny) * (module_y - module_overlap);
+      //! Note the module ordering is different for front and back side
+      xml_comp_t x_modCurr = iy % 2 == 0 ? x_modFront : x_modBack;
+
+      // double module_z = envelope.length() / 2.0 - total_thickness/2;
+      double module_z = x_supp_envelope.length() / 2.0 + total_thickness / 2;
+      if (iy % 2 == 0) {
+        module_z *= -1;
+      }
+      float corner_x1 = xcoord + module_x / 2;
+      float corner_x2 = xcoord - module_x / 2;
+      float corner_y1 = ycoord + module_y / 2;
+      float corner_y2 = ycoord - module_y / 2;
+      float maxRadius =
+          std::max(std::max(sqrt(pow(corner_x1, 2) + pow(corner_y1, 2)), sqrt(pow(corner_x2, 2) + pow(corner_y2, 2))),
+                   std::max(sqrt(pow(corner_x1, 2) + pow(corner_y2, 2)), sqrt(pow(corner_x2, 2) + pow(corner_y1, 2))));
+      float minRadius =
+          std::min(std::min(sqrt(pow(corner_x1, 2) + pow(corner_y1, 2)), sqrt(pow(corner_x2, 2) + pow(corner_y2, 2))),
+                   std::min(sqrt(pow(corner_x1, 2) + pow(corner_y2, 2)), sqrt(pow(corner_x2, 2) + pow(corner_y1, 2))));
```

Comment:
Consider `std::hypot(corner_x1, corner_y1)` etc instead of `sqrt(sum(pow()))`.

Reply:
Sure, changed to hypot.